### PR TITLE
[IMP] account: default taxes for multi-company product creation

### PR DESCRIPTION
**Pull request by alialfie**
Before:
When a new product was created in a multi-company setting, if the company field is empty, the product is available for all companies. The default sale and purchase tax of the company was set on the product. The problem is that only the default taxes of the currently  active company was set on the product, so viewing  the product in other companies showed an empty field for the tax.

Now:
When creating a new product with the company field empty, the default taxes of the other companies are set on the product as well.

task-3375286

Enterprise PR: odoo/enterprise/pull/45075

### CODE REVIEW

File: addons/account/models/product.py [hunk 1]
```diff
@@ -114,6 +114,35 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    def _get_default_company_taxes(self, companies):
+        default_taxes_ids = [company.account_sale_tax_id.id for company in companies if company.account_sale_tax_id]
+        default_supplier_taxes_ids = [company.account_purchase_tax_id.id for company in companies if company.account_purchase_tax_id]
+
+        return default_taxes_ids, default_supplier_taxes_ids
```
Comment (rugo-odoo) [inline addons/account/models/product.py:121]: Although, I like it (a lot) personally. I'm not sure we need to create a function for that as it is only use once (if I'm not mistaken).

Curious to see what @qdp-odoo thinks.

**Review by rugo-odoo — CHANGES_REQUESTED** [review]

File: addons/account/models/account_tax.py [hunk 2]
```diff
@@ -524,7 +524,7 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
         if not self:
             company = self.env.company
         else:
-            company = self[0].company_id._accessible_branches()[:1]
+            company = self[0].company_id._accessible_branches()[:1] or self[0].company_id
```
Comment (smetl) [inline addons/account/models/account_tax.py:527]: To me `_accessible_branches` should always return a result... :thinking:

File: addons/account/models/product.py [hunk 3]
```diff
@@ -118,6 +118,31 @@ def _check_uom_not_in_invoice(self):
...
+                            vals['taxes_id'] = vals['taxes_id'] + [Command.link(tax_id) for tax_id in default_taxes_ids] if vals['taxes_id'] else [Command.set(default_taxes_ids)]
+                        else:
+                            vals['taxes_id'] = [Command.set(default_taxes_ids + self.env.company.account_sale_tax_id.ids)]
+
+                    if self._context.get('default_supplier_taxes_id', True):
+                        if 'supplier_taxes_id' in vals:
+                            vals['supplier_taxes_id'] = vals['supplier_taxes_id'] + [Command.link(tax_id) for tax_id in default_supplier_taxes_ids] if vals['supplier_taxes_id'] else [Command.set(default_supplier_taxes_ids)]
+                        else:
+                            vals['supplier_taxes_id'] = [Command.set(default_supplier_taxes_ids + self.env.company.account_purchase_tax_id.ids)]
+        return super(ProductTemplate, self.sudo()).create(vals_list)
```
Comment (smetl) [inline addons/account/models/product.py:144]: To me, you could simplify this a lot by creating the product first, then add the extra taxes if necessary.
```suggestion
@api.model_create_multi
def create(self, vals_list):
    products = super().create(vals_list)

    if products.company_id:
        return products

    # If no company was set for the product, the product will be available for all companies and therefore should
    # have the default taxes of the other companies as well. sudo() is used since we're going to need to fetch all
    # the other companies default taxes which the user may not have access to.
    other_companies = self.env['res.company'].sudo().search([])
    if not other_companies:
        return products

    default_customer_taxes = other_companies.filtered('account_sale_tax_id').account_sale_tax_id
    default_supplier_taxes = other_companies.filtered('account_purchase_tax_id').account_purchase_tax_id

    for product in products:
        if product.taxes_id:
            product_customer_taxes = default_customer_taxes - product.company_id.account_sale_tax_id
        else:
            product_customer_taxes = default_customer_taxes
        product.taxes_id = [Command.link(tax.id) for tax in product_customer_taxes]

        if product.supplier_taxes_ids:
            product_supplier_taxes = default_supplier_taxes - product.company_id.account_purchase_tax_id
        else:
            product_supplier_taxes = default_supplier_taxes
        product.supplier_taxes_ids = [Command.link(tax.id) for tax in product_supplier_taxes]

    return products
```

File: addons/account/tests/common.py [hunk 4]
```diff
@@ -93,6 +94,23 @@ def setUpClass(cls, chart_template_ref=None):
...
+            'standard_price': 800.0,
+            'taxes_id': [Command.set(cls.tax_sale_a.ids)],
+            'supplier_taxes_id': [Command.set(cls.tax_purchase_a.ids)],
+        })
+        cls.product_without_company_b = cls.env['product.product'].create({
+            'name': 'product_without_company_b',
+            'uom_id': cls.env.ref('uom.product_uom_dozen').id,
+            'lst_price': 200.0,
+            'standard_price': 160.0,
+            'taxes_id': [Command.set((cls.tax_sale_a + cls.tax_sale_b).ids)],
```
Comment (smetl) [inline addons/account/tests/common.py:113]: Instead of changing the setup for everyone, I think it's better to add the test products directly on your test file

File: addons/account/tests/test_tax.py [hunk 5]
```diff
@@ -1168,3 +1168,20 @@ def test_parse_name_search(self):
...
+        product_with_company = self.env['product.template'].create({
+            'name': 'Product With a Company',
+            'company_id': self.company_data['company'].id
+        })
+        # Product should have all the default taxes of the other companies.
+        self.assertEqual(product_without_company.taxes_id, companies.account_sale_tax_id)
+        self.assertEqual(product_without_company.supplier_taxes_id, companies.account_purchase_tax_id)
+        # Product should have only the default tax of the company it belongs to.
+        self.assertEqual(product_with_company.taxes_id, self.company_data['company'].account_sale_tax_id)
+        self.assertEqual(product_with_company.supplier_taxes_id, self.company_data['company'].account_purchase_tax_id)
```
Comment (smetl) [inline addons/account/tests/test_tax.py:1187]: This file is more to test the taxes computation. Instead, create a new file like `test_product`.
Also, use `assertRecordValues` to test multiple field values at the same time

File: addons/account/tests/test_product.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            'company_id': cls.company_data['company'].id
+        })
+
+    def test_multi_company_product_tax(self):
+        """ Ensure default taxes are set for all companies on products with no company set. """
+        companies = self.env['res.company'].sudo().search([])
+        # Product should have all the default taxes of the other companies.
+        self.assertRecordValues(self.product_without_company, {
+            'taxes_id': companies.account_sale_tax_id,
+            'supplier_taxes_id': companies.account_purchase_tax_id})
```
Comment (smetl) [inline addons/account/tests/test_product.py:28]: The second parameter of `assertRecordValues` is a list
```suggestion
        self.assertRecordValues(self.product_without_company, [{
            'taxes_id': companies.account_sale_tax_id,
            'supplier_taxes_id': companies.account_purchase_tax_id,
        }])
```

**Comment by alialfie** [comment]
@rugo-odoo @smetl Changes are done!

File: addons/account/tests/test_product.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            'company_id': cls.company_data['company'].id
+        })
+
+    def test_multi_company_product_tax(self):
+        """ Ensure default taxes are set for all companies on products with no company set. """
+        companies = self.env['res.company'].sudo().search([])
+        # Product should have all the default taxes of the other companies.
+        self.assertRecordValues(self.product_without_company.sudo(), [{
+            'taxes_id': companies.account_sale_tax_id.ids,
+            'supplier_taxes_id': companies.account_purchase_tax_id.ids}])
```
Comment (rugo-odoo) [inline addons/account/tests/test_product.py:28]:
```suggestion
        self.assertRecordValues(self.product_without_company.sudo(), [{
            'taxes_id': companies.account_sale_tax_id.ids,
            'supplier_taxes_id': companies.account_purchase_tax_id.ids,
        }])
```
Formatting it like this allows others to add new lines without breaking the blame. (same for the next assert).

File: addons/account/tests/test_product.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        cls.env.user.company_id = cls.company_data['company']
+        cls.product_without_company = cls.env['product.template'].create({
+            'name': 'Product Without a Company',
+        })
+        cls.product_with_company = cls.env['product.template'].create({
+            'name': 'Product With a Company',
+            'company_id': cls.company_data['company'].id
+        })
```
Comment (rugo-odoo) [inline addons/account/tests/test_product.py:20]:
```suggestion
        cls.product_with_company = cls.env['product.template'].create({
            'name': 'Product With a Company',
            'company_id': cls.company_data['company'].id,
        })
```

File: addons/account/tests/test_product.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestAccountPaymentTerms(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
```
Comment (rugo-odoo) [inline addons/account/tests/test_product.py:10]: In this case, there is no need for this. You can do this on the test itself.

File: addons/project/tests/test_project_profitability.py [hunk 10]
```diff
@@ -39,7 +39,7 @@ def setUpClass(cls):
         cls.foreign_currency = cls.env['res.currency'].create({
             'name': 'Chaos orb',
             'symbol': '☺',
-            'rounding': 0.001,
+            'rounding': 0.01,
```
Comment (rugo-odoo) [inline addons/project/tests/test_project_profitability.py:42]: Why this change ?

**Review by rugo-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/tests/test_point_of_sale_flow.py [hunk 11]
```diff
@@ -405,10 +405,12 @@ def test_order_to_picking02(self):
             'type': 'product',
             'tracking': 'lot',
             'available_in_pos': True,
+            'company_id': self.company_data['company'].id,
```
Comment (Feyensv) [inline addons/point_of_sale/tests/test_point_of_sale_flow.py:408]: There is a lot of places where you add this `company_id` but is it always necessary ?
Isn't it hiding real problems where the taxes are not filtered by company, or the test/code does operations with the wrong company(ies) in the environment ?

File: addons/website_sale/tests/test_sale_process.py [hunk 12]
```diff
@@ -405,7 +405,7 @@ def test_07_change_fiscal_position(self):
             'list_price': 100,
             'website_published': True,
             'sale_ok': True,
-            'taxes_id': [tax_10_incl.id]
+            'taxes_id': [Command.set(tax_10_incl.ids)],
```
Comment (Feyensv) [inline addons/website_sale/tests/test_sale_process.py:408]: What exactly is making this change necessary ?
Either it wasn't working before, or I don't see what change you do that makes it not work anymore :thinking:.

**Review by Feyensv — COMMENTED** [review]
Shouldn't we also make sure that modifying the company on the product updates its taxes (or at least filters out invalid taxes from other companies) ?

File: addons/account/models/product.py [hunk 13]
```diff
@@ -118,6 +118,28 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        products = super().create(vals_list)
+        if products.company_id:
+            return products
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py:125]: That's a very good point that was handled in the old implementation but missed in the new one, thanks!

File: addons/account/models/product.py [hunk 14]
```diff
@@ -118,6 +118,28 @@ def _check_uom_not_in_invoice(self):
...
+
+        default_customer_tax_ids = other_companies.filtered('account_sale_tax_id').account_sale_tax_id.ids
+        default_supplier_tax_ids = other_companies.filtered('account_purchase_tax_id').account_purchase_tax_id.ids
+
+        for product in products.sudo():
+            if self._context.get('default_taxes_id', True):
+                product.taxes_id = [Command.link(tax) for tax in default_customer_tax_ids]
+            if self._context.get('default_supplier_taxes_id', True):
+                product.supplier_taxes_id = [Command.link(tax) for tax in default_supplier_tax_ids]
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py:140]: Correct me if I'm wrong but doesn't the context key `default_[field_name]=False` mean that we shouldn't get the default values for that field while creating a record? So the logic here actually just makes sure that the key value is not False if it exists, to make sure we can add the default taxes of the rest of the companies.



> self.env.context

What's the difference?

File: addons/point_of_sale/tests/test_point_of_sale_flow.py (hunk 11 above)
Comment (alialfie) [reply addons/point_of_sale/tests/test_point_of_sale_flow.py:408]: Indeed, this is a bug in the ORM and you can read more about it odoo/odoo/pull/132795. Until this is fixed, we resort to the `invalidate_model()` function for multi-company tests, and for tests not requiring a multi-company setup we add the company_id.

File: addons/website_sale/tests/test_sale_process.py (hunk 12 above)
Comment (alialfie) [reply addons/website_sale/tests/test_sale_process.py:408]: It made the older implementation cleaner and I decided to leave it in the commit even after the new implementation not needing it. Should I change it back?

File: addons/account/models/product.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/account/models/product.py:140]: Well, basically, both the `default_` key in the context set to a falsy value, and any write value given are expected to cancel the default behavior.
In this situation, I don't really know what could be the best approach honestly.  We could consider that we don't need to add taxes for any of the environment companies, and change the default to set all the taxes of the current companies :thinking:.

It's tricky to make sure that we do not add unexpected taxes that the customer wouldn't want.

File: addons/website_sale/tests/test_sale_process.py (hunk 12 above)
Comment (Feyensv) [reply addons/website_sale/tests/test_sale_process.py:408]: I don't mind, but I was curious why this change was done ;)

File: addons/account/models/product.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py:140]: > We could consider that we don't need to add taxes for any of the environment companies, and change the default to set all the taxes of the current companies

Not sure I understand, can you elaborate more please?

[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/account/models/product.py]: Currently, on the field, the default is `lambda self: self.env.company.account_sale_tax_id or self.env.company.root_id.account_sale_tax_id`.

We could say that if you have multiple companies enabled in the company switcher, we add the taxes for all the companies by default: `lambda self: self.env.companies.account_sale_tax_id or self.env.companies.root_id.account_sale_tax_id`, and in your create override, you would only add the defaults for companies not chosen in the switcher
`other_companies = self.env['res.company'].sudo().search([('id', 'not in', self.env.companies.ids)])`

File: addons/project/tests/test_project_profitability.py (hunk 10 above)
Comment (alialfie) [reply addons/project/tests/test_project_profitability.py:42]: This [test](https://runbot.odoo.com/runbot/build/50027558) failed due to rounding inconsistency. This fixes it.

File: addons/account/models/product.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py]: I had already discussed this with @chklop and it was decided not to change the default on the field as it would over-complicate things and also having multiple companies enabled in the company switcher is not going to be super common :sweat_smile:

[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/account/models/product.py]: I think multi-company is intrinsically a complex thing and half magic solutions may lead to more problems than they solve, but well, let's start with the simple solution then.

**Comment by alialfie** [comment]
@Feyensv I've added the changes you mentioned, thanks again!

@rugo-odoo Should be ready for a final review now, all yours :smile:

**Comment by rugo-odoo** [comment]
LGTM

File: addons/account/models/account_tax.py (hunk 2 above)
Comment (william-andre) [reply addons/account/models/account_tax.py:533]: If it doesn't return something, it means that you would have an access error when trying to read `self[0].company_id` anyways...

File: addons/project/tests/test_project_profitability.py (hunk 10 above)
Comment (qdp-odoo) [reply addons/project/tests/test_project_profitability.py:42]: hum yeah.. but why was the test failing because of your changes?

File: addons/account/tests/test_product.py [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+        })
+        companies = self.env['res.company'].sudo().search([])
+        # Product should have all the default taxes of the other companies.
+        self.assertRecordValues(product_without_company.sudo(), [{
+            'taxes_id': companies.account_sale_tax_id.ids,
+            'supplier_taxes_id': companies.account_purchase_tax_id.ids,
+        }])
+        # Product should have only the default tax of the company it belongs to.
+        self.assertRecordValues(product_with_company, [{
+            'taxes_id': self.company_data['company'].account_sale_tax_id.ids,
```
Comment (qdp-odoo) [inline addons/account/tests/test_product.py:30]: to be fair, the record tested here should be browsed in sudo(), otherwise the orm could filter some set taxes you shouldn't see, no?

File: addons/account/tests/common.py [hunk 16]
```diff
@@ -92,6 +94,8 @@ def setUpClass(cls, chart_template_ref=None):
             'standard_price': 160.0,
             'property_account_income_id': cls.copy_account(cls.company_data['default_account_revenue']).id,
             'property_account_expense_id': cls.copy_account(cls.company_data['default_account_expense']).id,
+        })
+        cls.product_b.sudo().write({
```
Comment (qdp-odoo) [inline addons/account/tests/common.py:98]: idem. why the sudo?

File: addons/hr_expense/tests/test_expenses_mail_import.py [hunk 17]
```diff
@@ -59,6 +59,7 @@ def test_import_expense_from_email_product_no_cost(self):
             'standard_price': 0.0,
             'can_be_expensed': True,
             'default_code': 'product_no_cost',
+            'company_id': self.company_data['company'].id,
```
Comment (qdp-odoo) [inline addons/hr_expense/tests/test_expenses_mail_import.py:62]: instead, don't we wanna set a default company on product_template? @chklop 

 feels like a lot of useless diff

File: addons/sale/tests/common.py [hunk 18]
```diff
@@ -170,6 +170,7 @@ def setup_sale_configuration_for_company(cls, company):
                 'expense_policy': 'sales_price',
                 'taxes_id': [(6, 0, [])],
                 'supplier_taxes_id': [(6, 0, [])],
+                # 'company_id': company.id,
```
Comment (qdp-odoo) [inline addons/sale/tests/common.py:173]: dead code

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
questions, questions and more questions :)

File: addons/hr_expense/tests/test_expenses_mail_import.py (hunk 17 above)
Comment (chklop) [reply addons/hr_expense/tests/test_expenses_mail_import.py:62]: By default a product should be create on all companies and not restricted to one, I don't want to change that behaviour in Odoo.

File: addons/account/tests/common.py [hunk 19]
```diff
@@ -92,6 +90,13 @@ def setUpClass(cls, chart_template_ref=None):
             'standard_price': 160.0,
             'property_account_income_id': cls.copy_account(cls.company_data['default_account_revenue']).id,
             'property_account_expense_id': cls.copy_account(cls.company_data['default_account_expense']).id,
+        })
+        cls.env['product.template'].invalidate_model()
```
Comment (qdp-odoo) [inline addons/account/tests/common.py:94]: why do you need to invalidate_model()?

File: addons/account/tests/test_product.py [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestAccountPaymentTerms(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
```
Comment (qdp-odoo) [inline addons/account/tests/test_product.py:11]: setUpClass exactly the same as parent, so it's unnecessary to provide it?

File: addons/account/tests/common.py [hunk 21]
```diff
@@ -92,6 +90,13 @@ def setUpClass(cls, chart_template_ref=None):
...
+        cls.product_a.write({
+            'taxes_id': [Command.set(cls.tax_sale_a.ids)],
+            'supplier_taxes_id': [Command.set(cls.tax_purchase_a.ids)],
+        })
+        cls.product_b.write({
```
Comment (qdp-odoo) [inline addons/account/tests/common.py:99]: any reason why you do that in write() instead of directly during the create(), few lines above? :thinking:

File: addons/project/tests/test_project_profitability.py (hunk 10 above)
Comment (qdp-odoo) [reply addons/project/tests/test_project_profitability.py:42]: hum... more taxes added? I suppose you refer to the taxes of other companies, but that shouldn't change the computation in the context of the current user's company?

to me,  that still looks like a patch hiding the real problem/bug

File: addons/purchase/tests/test_purchase.py [hunk 22]
```diff
@@ -382,6 +384,8 @@ def test_supplier_list_in_product_with_multicompany(self):
         product = self.env['product.product'].create({
             'name': 'product_test',
         })
+        self.env['product.template'].invalidate_model()
```
Comment (qdp-odoo) [inline addons/purchase/tests/test_purchase.py:387]: why the model invalidation?

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
I've more questions, actually ^^'

**Review by william-andre — COMMENTED** [review]
I think the issue raised by the spec is a bit unsolvable :confused:

File: addons/account/models/product.py [hunk 23]
```diff
@@ -118,6 +118,27 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    @api.model_create_multi
+    def create(self, vals_list):
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py:122]: Discussed both of those cases with @chklop and it was decided that those cases should not be handled. For the first case, we simply only want to add those taxes during creation as it is mentioned in the UI that default taxes will apply to newly created products. For the second case, what if you had changed some of the tax values for the product in some companies, added a company id then later removed it? You'd lose those specific tax values. I see your point though, but again, chkl said we don't care about some extra rows in the database.

File: addons/account/models/product.py [hunk 24]
```diff
@@ -118,6 +118,27 @@ def _check_uom_not_in_invoice(self):
...
+        other_companies = self.env['res.company'].sudo().search([('id', '!=', self.env.company.id)])
+        if not other_companies:
+            return products
+
+        default_customer_tax_ids = other_companies.filtered('account_sale_tax_id').account_sale_tax_id.ids
+        default_supplier_tax_ids = other_companies.filtered('account_purchase_tax_id').account_purchase_tax_id.ids
+
+        for product in products.sudo():
+            if not product.company_id:
+                if self.env.context.get('default_taxes_id', True):
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py:136]: > Why are we checking the context?
> 
To have the same behaviour as the orm when it comes to `default_` context keys
> Note that the ORM team is working on removing the `default_` context keys
> 
I did not know that. Are they providing an alternative way then?
> Also, what happens if the tax values were set in the `vals`? Do we still want the default taxes on the other companies in this case?
>
Yes, since in the UI you're only seeing the taxes of the current company you're using, and if you're adding non default taxes they will stay the same for that company but for the rest of the companies, you'll get the default taxes.

File: addons/account/models/product.py [hunk 25]
```diff
@@ -118,6 +118,27 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        products = super().create(vals_list)
+        # If no company was set for the product, the product will be available for all companies and therefore should
+        # have the default taxes of the other companies as well. sudo() is used since we're going to need to fetch all
+        # the other companies default taxes which the user may not have access to.
+        other_companies = self.env['res.company'].sudo().search([('id', '!=', self.env.company.id)])
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/account/models/product.py:127]: You're probably right, I started this task before you did the company branches feature. I'll take a look, thanks!

File: addons/project/tests/test_project_profitability.py (hunk 10 above)
Comment (qdp-odoo) [reply addons/project/tests/test_project_profitability.py:42]: instead, you must patch the failing assert and check 'billed' field as `expense_foreign.currency_id.round(-280.0 - expense_foreign.untaxed_amount * 0.2)`, I suppose... which should be equal -340.87

to be discussed, live, tomorrow. :)

File: addons/l10n_ar/demo/product_product_demo.xml [hunk 26]
```diff
@@ -7,6 +7,7 @@
         <field name="default_code">AFIP_DESPACHO</field>
         <field name="sale_ok" eval="False"/>
         <field name="uom_id" ref="uom.product_uom_unit"/>
+        <field name="company_id" ref="company_ri"/>
```
Comment (william-andre) [inline addons/l10n_ar/demo/product_product_demo.xml:10]: Pretty sure these products need to be usable by all the AR demo companies (RI, Mono and Exento)

**Review by xavierbol — APPROVED** [review]

**Review by william-andre — APPROVED** [review]
Looking a lot better now! :+1:

**Comment by william-andre** [comment]
@robodoo r+
